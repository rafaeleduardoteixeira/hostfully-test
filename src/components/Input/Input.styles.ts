import styled from 'styled-components';
import { colors, media } from '../../themes';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

export const InputLabel = styled.label`
  font-size: 1rem;
  font-weight: 500;
  margin-bottom: 5px;
  color: ${colors.grayDark};
`;

export const InputField = styled.input<{ width?: string; error?: string }>`
  height: 50px;
  border: none;
  outline: none;
  padding: 0 10px 0 20px;
  font-size: 1rem;
  color: ${colors.black};
  background-color: ${colors.grayInput};
  border-radius: 10px;
  width: ${(props) => props.width || '90%'};

  outline: ${(props) => (props.error ? `1px solid ${colors.red}` : 'none')};

  &::placeholder {
    color: ${colors.gray};
  }

  &:focus {
    outline: ${colors.grayLight} auto 1px;
  }

  &:disabled {
    cursor: not-allowed;
  }

  @media (${media.mobile}) {
    font-size: 0.9rem;
    height: 40px;
  }
`;
