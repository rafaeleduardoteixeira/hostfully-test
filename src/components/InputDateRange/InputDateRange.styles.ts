import styled from 'styled-components';
import { colors } from '../../themes';

export const Container = styled.div<{ error?: string }>`
  .react-datepicker__input-container {
    display: flex;
    align-items: center;
    img {
      right: 10px;
    }
  }
  .react-datepicker__day--in-range {
    background-color: ${colors.hostfullyPrimaryColor} !important;
  }

  .react-datepicker {
    font-size: 1rem;
    border: none;
    border-radius: 10px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  }

  .react-datepicker__header {
    background-color: ${colors.hostfullySecondaryColor};
  }

  .react-datepicker__current-month {
    color: white;
  }

  .react-datepicker__day-name {
    color: white;
  }

  .react-datepicker__triangle {
    display: none !important;
  }

  .date-picker {
    height: 50px;
    border-radius: 10px;
    background-color: ${colors.grayInput};
    border: none;
    color: ${colors.gray};
    padding: 0 20px;
    width: 290px;
    font-size: 1rem;

    outline: ${(props) => (props.error ? `1px solid ${colors.red}` : 'none')};

    &::placeholder {
      color: ${colors.gray};
    }
    &:focus {
      outline: ${colors.grayLight} auto 1px;
    }
  }
`;
