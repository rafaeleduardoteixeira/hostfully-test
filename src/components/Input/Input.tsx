import React, { ChangeEvent } from 'react';
import { Container, InputLabel, InputField } from './Input.styles';

interface InputProps {
  name?: string;
  label?: string;
  value?: string;
  type?: string;
  placeholder?: string;
  width?: string;
  disabled?: boolean;
  error?: string;
  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
}

export const Input = (props: InputProps) => {
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    props.onChange && props.onChange(event);
  };

  return (
    <Container>
      {props.label && <InputLabel htmlFor={props.name}>{props.label}</InputLabel>}
      <InputField data-testid="Input" {...props} onChange={handleChange} />
    </Container>
  );
};
