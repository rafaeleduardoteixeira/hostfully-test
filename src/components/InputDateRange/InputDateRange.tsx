import moment from 'moment';
import { Container } from './InputDateRange.styles';

//Calendar
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

interface InputDateRangeProps {
  bookedDates?: {
    start: Date | null;
    end: Date | null;
  }[];
  dateStart: Date | undefined;
  dateEnd: Date | undefined;
  error?: string;
  onChange: (value: any) => void;
}

export const InputDateRange = ({ bookedDates, dateStart, dateEnd, error, onChange }: InputDateRangeProps) => {
  const excludeDateIntervals = bookedDates?.map((bookedDate) => {
    return {
      start: moment(bookedDate.start, 'America/Los_Angeles').subtract(1, 'days').toDate(),
      end: moment(bookedDate.end, 'America/Los_Angeles').toDate(),
    };
  }) as { start: Date; end: Date }[];

  const handleChange = (value: any) => {
    if (value) {
      onChange(value);
      return;
    }
    onChange(null);
  };

  return (
    <Container error={error}>
      <DatePicker
        showIcon
        icon={<img src="/icons/calendar.svg" alt="calendar" />}
        className="date-picker"
        placeholderText="Select a date to book"
        dateFormat={'MMMM d, yyyy'}
        selected={dateStart}
        selectsRange
        startDate={dateStart}
        endDate={dateEnd}
        onChange={handleChange}
        minDate={new Date()}
        excludeDateIntervals={excludeDateIntervals}
        shouldCloseOnSelect={true}
      />
    </Container>
  );
};
