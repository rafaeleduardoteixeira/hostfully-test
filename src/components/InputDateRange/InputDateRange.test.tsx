import React, { act } from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { InputDateRange } from './InputDateRange';
import { Input } from '../Input/Input';

describe('InputDateRange', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(<InputDateRange onChange={jest.fn()} dateStart={undefined} dateEnd={undefined} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Should render InputDateRange', () => {
    render(
      <InputDateRange
        onChange={jest.fn()}
        dateStart={undefined}
        dateEnd={undefined}
        bookedDates={[{ start: new Date(), end: new Date() }]}
      />,
    );
    const InputDateRangeElement = screen.getByPlaceholderText('Select a date to book');
    expect(InputDateRangeElement).toBeInTheDocument();
  });

  test('Should select a date', () => {
    const onChange = jest.fn();
    render(
      <InputDateRange
        onChange={onChange}
        dateStart={undefined}
        dateEnd={undefined}
        bookedDates={[{ start: new Date(), end: new Date() }]}
      />,
    );
    const InputDateRangeElement = screen.getByPlaceholderText('Select a date to book');
    act(() => {
      fireEvent.change(InputDateRangeElement, { target: { value: new Date() } });
    });
    expect(onChange).toHaveBeenCalled();
  });
});
