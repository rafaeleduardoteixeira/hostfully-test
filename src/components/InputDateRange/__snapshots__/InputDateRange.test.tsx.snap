// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDateRange Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu jKTTGs"
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container react-datepicker__view-calendar-icon"
      >
        <img
          alt="calendar"
          class=" react-datepicker__calendar-icon "
          src="/icons/calendar.svg"
        />
        <input
          class="date-picker"
          placeholder="Select a date to book"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
